### PR TITLE
Cosmos SDK v47 integration

### PR DESCRIPTION
**Pull request by JayT106**
Closes: #1667

## Description

integrate ethermint with Cosmos SDK V47
______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

**Comment by JayT106** [comment]
when running the target `make tools` will remove `cosmossdk.io/simapp` and `github.com/cosmos/ibc-go/v7 v7.0.0-rc0`. Not sure how to fix it.

**Comment by thomas-nguy** [comment]
I think some protos still have single word name for the`cosmos_proto.implements_interface` annotation
https://github.com/cosmos/cosmos-sdk/blob/release/v0.47.x/UPGRADING.md#acceptsimplements_interface-proto-annotations

**Comment by thomas-nguy** [comment]
for generating proto files with `proto gen` shouldnt you use `ghcr.io/cosmos/proto-builder` image (version >= 0.11.5) ?

**Comment by thomas-nguy** [comment]
you can probably use the same protoImage for all proto related command like there
https://github.com/cosmos/cosmos-sdk/blob/v0.47.0-rc3/Makefile#L412

File: go.mod [hunk 1]
```diff
@@ -163,37 +171,45 @@ require (
...
 	github.com/99designs/keyring => github.com/cosmos/keyring v1.1.7-0.20210622111912-ef00f8ac3d76
+
+	// github.com/miguelmota/go-ethereum-hdwallet is still importing github.com/btcsuite/btcutil/hdkeychain
+	github.com/btcsuite/btcutil => github.com/btcsuite/btcd/btcutil v1.1.1
+
 	// Fix upstream GHSA-h395-qcrw-5vmq vulnerability.
 	// TODO Remove it: https://github.com/cosmos/cosmos-sdk/issues/10409
 	github.com/gin-gonic/gin => github.com/gin-gonic/gin v1.7.0
 	// use cosmos flavored protobufs
 	github.com/gogo/protobuf => github.com/regen-network/protobuf v1.3.3-alpha.regen.1
```
Comment (thomas-nguy) [inline go.mod:211]: can it be removed?

Comment (JayT106) [reply go.mod:211]: Unless we fork ' go-ethereum-hdwallet'. It is still using `github.com/btcsuite/btcutil`

**Comment by JayT106** [comment]
> you can probably use the same protoImage for all proto related command like there https://github.com/cosmos/cosmos-sdk/blob/v0.47.0-rc3/Makefile#L412

thanks, updated!

File: go.mod (hunk 1 above)
Comment (JayT106) [reply go.mod:211]: @yihuang has a fork for solving the issue, maybe we can use this instead?
https://github.com/crypto-org-chain/go-ethereum-hdwallet/commit/1a13de758d2fbb965bc393bca3844b9204cab836

**Review by GAtom22 — CHANGES_REQUESTED** [review]
Thanks @JayT106! Great work!
Could you please address the conflicts and failing checks?

**Pull request closed without merging**